### PR TITLE
fetch rotated task logs on failure

### PR DESCRIPTION
**Pull request by falseperfection**
https://jira.mesosphere.com/browse/INFINITY-2249

### CODE REVIEW

**Comment by falseperfection** [comment]
Agreed @mellenburg. I can put the `*.log => task_logs.tgz` line back in the TeamCity config to handle compression there. I don't see a way to compress each file individually, without knowing the filenames ahead of time.

**Comment by benclarkwood** [comment]
The only log that will inflate much is the scheduler log (we don't do anything to the tasks such that they'd write a lot of logs)

I'd expect at most 20MB of logs for a failed test.

File: conftest.py [hunk 1]
```diff
@@ -64,20 +64,33 @@ def pytest_runtest_makereport(item, call):
...
 
 
-@pytest.fixture(autouse=True)
-def get_task_logs_on_failure(request):
-    """ Scheduler should be the only task running as root
-    """
-    yield
+def get_rotating_task_log_lines(task_id: str, task_file: str):
+    rotated_filenames = [task_file,]
+    rotated_filenames.extend(['{}.{}'.format(task_file, i) for i in range(1, 10)])
```
Comment (elezar) [inline conftest.py]: Is there a reason that we limit this to `9`?  (We probably won't need more than this depending on the log rotation frequency though).

Comment (elezar) [inline conftest.py]: Just a thought, isn't this going to append the logs out of order? The file with no enumeration contains the most recent logs, but this is being returned first, followed by the oldest(?) logs?

File: conftest.py [hunk 2]
```diff
@@ -64,20 +64,33 @@ def pytest_runtest_makereport(item, call):
...
+    yield
+    if is_test_failure(request):
         for task_id in get_task_ids():
             for task_file in ('stderr', 'stdout'):
-                log_name = '{}_{}_{}.log'.format(request.node.name, task_id, task_file)
-                task_logs = get_task_logs_for_id(task_id, task_file=task_file)
-                if task_logs:
+                for log_lines in get_rotating_task_log_lines(task_id, task_file):
+                    log_name = '{}_{}_{}.log'.format(request.node.name, task_id, task_file)
                     with open(log_name, 'w') as f:
```
Comment (elezar) [inline conftest.py]: Here we probably want to use `open(log_name, 'a')` instead. It is my understanding from: https://docs.python.org/3/library/functions.html#open that `open(log_name, 'w')` will clear the file before writing to it.

File: conftest.py (hunk 1 above)
Comment (falseperfection) [inline conftest.py]: The lines aren't appended, each log file is downloaded as a separate file. `9` was just the number in the ticket. If we find we need more we can change it, or, perhaps, add a call to look up how many rotated log files already exist.

File: conftest.py (hunk 2 above)
Comment (falseperfection) [inline conftest.py]: You're right that `'a'` would append. My intent was to open a separate file for each task log. So the names will look like: `<node.name>_<task_id>_stdout.log`, `<node.name>_<task_id>_stdout.1.log`, etc. Good catch!

**Comment by benclarkwood** [comment]
@elezar Logrotate only leaves 9 rotate files.

File: conftest.py (hunk 1 above)
Comment (elezar) [reply conftest.py]: @benclarkwood said that there are only 9 files, so this is fine.

**Comment by elezar** [comment]
@benclarkwood @falseperfection can we delete this branch?

**Comment by benclarkwood** [comment]
si

**Pull request merged**